core/parachain/pvf/pvf_impl.cpp

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "parachain/pvf/pvf_impl.hpp"
+
+#include "runtime/common/executor.hpp"
+#include "runtime/common/uncompress_code_if_needed.hpp"
+#include "runtime/module.hpp"
+#include "runtime/runtime_code_provider.hpp"
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::parachain, PvfError, e) {
+  using kagome::parachain::PvfError;
+  switch (e) {
+    case PvfError::NO_PERSISTED_DATA:
+      return "PvfError::NO_PERSISTED_DATA";
+    case PvfError::POV_SIZE:
+      return "PvfError::POV_SIZE";
+    case PvfError::POV_HASH:
+      return "PvfError::POV_HASH";
+    case PvfError::CODE_HASH:
+      return "PvfError::CODE_HASH";
+    case PvfError::SIGNATURE:
+      return "PvfError::SIGNATURE";
+    case PvfError::HEAD_HASH:
+      return "PvfError::HEAD_HASH";
+    case PvfError::COMMITMENTS_HASH:
+      return "PvfError::COMMITMENTS_HASH";
+    case PvfError::OUTPUTS:
+      return "PvfError::OUTPUTS";
+  }
+  return fmt::format("PvfError({})", e);
+}
+
+namespace kagome::parachain {
+  using common::Hash256;
+  using network::HeadData;
+  using network::OutboundHorizontal;
+  using network::ParachainBlock;
+  using network::ParachainRuntime;
+  using network::UpwardMessage;
+  using primitives::BlockNumber;
+  using runtime::PersistedValidationData;
+
+  struct DontProvideCode : runtime::RuntimeCodeProvider {
+    outcome::result<gsl::span<const uint8_t>> getCodeAt(
+        const storage::trie::RootHash &) const override {
+      abort();
```

Comment:
Hm, where's it supposed to be used?

Reply:
This interface implementation is required, because other class `assert`s that pointer is not null.  
Method is not used.